src/Nixfmt/Pretty.hs

```diff
@@ -301,23 +301,32 @@ isIndented parts =
           nonEmpty _                  = True
           inits = map textInit $ filter nonEmpty parts
 
-isEmptyLine :: [StringPart] -> Bool
-isEmptyLine []           = True
-isEmptyLine [TextPart t] = Text.null $ Text.strip t
-isEmptyLine _            = False
+-- | If the last line has at least one space but nothing else, it cannot be
+-- cleanly represented in an indented string.
+lastLineIsSpaces :: [[StringPart]] -> Bool
+lastLineIsSpaces [] = False
+lastLineIsSpaces xs = case last xs of
+    [TextPart t] -> isSpaces t
+    _            -> False
+
+isInvisibleLine :: [StringPart] -> Bool
+isInvisibleLine []           = True
+isInvisibleLine [TextPart t] = Text.null $ Text.strip t
```

Comment:
Why not `isSpaces`, like in the rest of the code?

Reply:
This function is specifically to catch an edge case where strings are better represented as a simple `""` string, rather than an indented `''''` string. It is not about matching the behavior of Nix. I added this because nixfmt used to turn things like `"\n\n"` and `"\n  "` into indented strings.